instrumentation/excon/lib/opentelemetry/instrumentation/excon/middlewares/tracer_middleware.rb

```diff
@@ -71,43 +75,32 @@ def self.around_default_stack
           private
 
           def handle_response(datum)
-            if datum.key?(:otel_span)
-              datum[:otel_span].tap do |span|
-                return span if span.end_timestamp
+            datum.delete(:otel_span)&.tap do |span|
+              return span if span.end_timestamp
 
-                if datum.key?(:response)
-                  response = datum[:response]
-                  span.set_attribute('http.status_code', response[:status])
-                  span.status = OpenTelemetry::Trace::Status.error unless (100..399).include?(response[:status].to_i)
-                end
+              if datum.key?(:response)
+                response = datum[:response]
+                span.set_attribute(OpenTelemetry::SemanticConventions::Trace::HTTP_STATUS_CODE, response[:status])
+                span.status = OpenTelemetry::Trace::Status.error unless (100..399).include?(response[:status].to_i)
+              end
 
-                span.status = OpenTelemetry::Trace::Status.error("Request has failed: #{datum[:error]}") if datum.key?(:error)
+              span.status = OpenTelemetry::Trace::Status.error("Request has failed: #{datum[:error]}") if datum.key?(:error)
```

Comment:
Also not related to your PR but do you know if `error` is an Exception object? If so then it should probably recorded using `record_exception`?

Reply:
Yes, it is an exception: https://github.com/excon/excon/blob/v0.104.0/lib/excon/connection.rb#L308C20-L308C20

Reply:
Done